subgraphs/compound-forks/protocols/dforce/src/mapping.ts

```diff
@@ -0,0 +1,604 @@
+  event: NewLiquidationIncentive
+): void {
+  let protocol = getOrCreateProtocol();
+  let newLiquidationIncentiveMantissa =
+    event.params.newLiquidationIncentiveMantissa;
+  _handleNewLiquidationIncentive(protocol, newLiquidationIncentiveMantissa);
+}
+
+// toggle market.isActive based on pause status of mint/redeem/transfer
+export function handleMintPaused(event: MintPaused): void {
+  let marketID = event.params.iToken.toHexString();
+  let market = Market.load(marketID);
+  if (market == null) {
+    log.warning("[handleNewReserveFactor] Market not found: {}", [marketID]);
+    return;
+  }
+
+  let _marketStatus = getOrCreateMarketStatus(marketID);
+
+  _marketStatus.mintPaused = event.params.paused;
+  // isActive = false if any one of mint/redeem/transfer is paused
+  market.isActive = !anyTrue([
+    _marketStatus.mintPaused,
+    _marketStatus.redeemPaused,
+    _marketStatus.transferPaused,
+  ]);
+
+  market.save();
+  _marketStatus.save();
+}
+
+// toggle market.isActive based on pause status of mint/redeem/transfer
+export function handleRedeemPaused(event: RedeemPaused): void {
+  let marketID = event.params.iToken.toHexString();
+  let market = Market.load(marketID);
+  if (market == null) {
+    log.warning("[handleNewReserveFactor] Market not found: {}", [marketID]);
+    return;
+  }
+
+  let _marketStatus = getOrCreateMarketStatus(marketID);
+
+  _marketStatus.redeemPaused = event.params.paused;
+  // isActive = false if any one of mint/redeem/transfer is paused
+  market.isActive = !anyTrue([
+    _marketStatus.mintPaused,
+    _marketStatus.redeemPaused,
+    _marketStatus.transferPaused,
+  ]);
+
+  market.save();
+  _marketStatus.save();
+}
+
+// toggle market.isActive based on pause status of mint/redeem/transfer
+// transfer pause stops transfer of all iTokens
+export function handleTransferPaused(event: TransferPaused): void {
+  let protocol = getOrCreateProtocol();
+  let markets = protocol._marketIDs;
+  for (let i = 0; i < markets.length; i++) {
+    let marketID = markets[i];
+    let market = Market.load(marketID);
+    if (market == null) {
+      log.warning("[handleNewReserveFactor] Market not found: {}", [marketID]);
+      return;
+    }
+
+    let _marketStatus = getOrCreateMarketStatus(marketID);
+    _marketStatus.transferPaused = event.params.paused;
+    // isActive = false if any one of mint/redeem/transfer is paused
+    market.isActive = !anyTrue([
+      _marketStatus.mintPaused,
+      _marketStatus.redeemPaused,
+      _marketStatus.transferPaused,
+    ]);
+
+    market.save();
+    _marketStatus.save();
+  }
+}
+
+export function handleBorrowPaused(event: BorrowPaused): void {
+  // toggle market.canBorrowFrom based on BorrowPaused event
+  let marketId = event.params.iToken.toHexString();
+  let market = Market.load(marketId);
+  if (market != null) {
+    market.canBorrowFrom = !event.params.paused;
+    market.save();
+  } else {
+    log.warning("Market {} does not exist.", [marketId]);
```

Comment:
add "[handleBorrowPaused]" to be consistent

Reply:
fixed